<!DOCTYPE html>
<html lang="en"><head>
<script src="2023_11_pss_rnaseq_files/libs/clipboard/clipboard.min.js"></script>
<script src="2023_11_pss_rnaseq_files/libs/quarto-html/tabby.min.js"></script>
<script src="2023_11_pss_rnaseq_files/libs/quarto-html/popper.min.js"></script>
<script src="2023_11_pss_rnaseq_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="2023_11_pss_rnaseq_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="2023_11_pss_rnaseq_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="2023_11_pss_rnaseq_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.2.256">

  <meta name="author" content="Jessica Minnier, Associate Professor of Biostatistics">
  <title>Power and Sample Size for RNA-seq Experiments</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="2023_11_pss_rnaseq_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="2023_11_pss_rnaseq_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #767676;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #767676;  padding-left: 4px; }
    div.sourceCode
      { color: #545454; background-color: #fefefe; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #545454; } /* Normal */
    code span.al { color: #7928a1; } /* Alert */
    code span.an { color: #696969; } /* Annotation */
    code span.at { color: #a55a00; } /* Attribute */
    code span.bn { color: #7928a1; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #d91e18; } /* ControlFlow */
    code span.ch { color: #008000; } /* Char */
    code span.cn { color: #d91e18; } /* Constant */
    code span.co { color: #696969; } /* Comment */
    code span.cv { color: #696969; font-style: italic; } /* CommentVar */
    code span.do { color: #696969; font-style: italic; } /* Documentation */
    code span.dt { color: #7928a1; } /* DataType */
    code span.dv { color: #7928a1; } /* DecVal */
    code span.er { color: #7928a1; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #a55a00; } /* Float */
    code span.fu { color: #06287e; } /* Function */
    code span.im { } /* Import */
    code span.in { color: #696969; } /* Information */
    code span.kw { color: #d91e18; } /* Keyword */
    code span.op { color: #00769e; } /* Operator */
    code span.ot { color: #d91e18; } /* Other */
    code span.pp { color: #7928a1; } /* Preprocessor */
    code span.sc { color: #00769e; } /* SpecialChar */
    code span.ss { color: #008000; } /* SpecialString */
    code span.st { color: #008000; } /* String */
    code span.va { color: #a55a00; } /* Variable */
    code span.vs { color: #008000; } /* VerbatimString */
    code span.wa { color: #696969; font-style: italic; } /* Warning */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <link rel="stylesheet" href="2023_11_pss_rnaseq_files/libs/revealjs/dist/theme/quarto.css" id="theme">
  <link href="2023_11_pss_rnaseq_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="2023_11_pss_rnaseq_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="2023_11_pss_rnaseq_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="2023_11_pss_rnaseq_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-captioned.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-captioned) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-captioned.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-captioned .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-captioned .callout-caption  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-captioned.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-captioned.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-caption {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-caption {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-caption {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-captioned .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-captioned .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-captioned) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-caption {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-caption {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-caption {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-caption {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-caption {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Power and Sample Size for RNA-seq Experiments</h1>
  <p class="subtitle">OCTRI BERD Research Forum</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Jessica Minnier, Associate Professor of Biostatistics 
</div>
        <p class="quarto-title-affiliation">
            OHSU
          </p>
    </div>
</div>

  <p class="date">11/1/23</p>
</section>
<section id="goals" class="slide level2">
<h2>Goals</h2>
<ul>
<li><p>Review power and sample size statistical definitions</p>
<ul>
<li>Power, sample size, effect size, type I error, type II error</li>
</ul></li>
<li><p>Overview of RNA-seq data generation</p></li>
<li><p>Required components for power calculation</p></li>
<li><p>Overview of tools available</p></li>
<li><p>Examples of power calculations</p></li>
<li><p>Brief overview of more complex designs and scRNA-seq issues</p></li>
</ul>
</section>
<section id="prerequisites" class="slide level2">
<h2>Prerequisites</h2>
<ul>
<li><p>Some knowledge of basic power and sample size calculations and concepts</p>
<ul>
<li><a href="https://www.ohsu.edu/octri/octri-research-forum-your-monthly-clinical-and-translational-research-event">PSS 101 from BERD</a>, worth reviewing and watching!</li>
</ul></li>
<li><p>Familiarity with RNA-seq experiments; good overview: <span class="citation" data-cites="conesa2016survey">(<a href="#/thank-you" role="doc-biblioref" onclick="">Conesa et al. 2016</a>)</span></p></li>
<li><p>Familiarity with statistical models and concepts such as regression, p-values, probability distributions (some review in context of RNA-seq: <a href="https://github.com/hbctraining">Harvard Chan Bioinformatics Core (HBC) training</a>’s <a href="https://hbctraining.github.io/DGE_workshop_salmon_online/lessons/01c_RNAseq_count_distribution.html">DGE</a> and <a href="https://github.com/hbctraining/DGE_workshop_salmon_online/blob/master/lessons/05a_hypothesis_testing.md">hypothesis testing</a> lessons)</p></li>
<li><p>Preferred: some experience with R/Bioconductor</p></li>
</ul>
</section>
<section id="rna-seq" class="slide level2">
<h2>RNA-seq</h2>
<ul>
<li><p>Typically, transcriptomics: gene expression profiling of samples, using next-generation sequencing methods, reference genome</p></li>
<li><p>RNA-seq can also examine alternative gene spliced transcripts, mutations, levels of other RNA besides messenger RNA, etc, but common aim is to evaluate gene expression measured by mRNA observed in sample tissue</p></li>
<li><p>“Bulk RNA-seq” to distinguish from single cell RNA-seq, large populations of cells, mixed cell types</p></li>
<li><p>Alternative method: microarrays, some of the following concepts still apply but distribution of expression measures is not count-based, but more related to continuous normal distribution <span class="citation" data-cites="lee2002power">(<a href="#/thank-you" role="doc-biblioref" onclick="">Lee and Whitmore 2002</a>)</span></p></li>
</ul>
</section>
<section id="rna-seq-data-generation" class="slide level2 smaller">
<h2>RNA-seq data generation</h2>
<p>NGS or massively parallel sequencing: sample preparation, mRNA fragmentation, reverse transcription to complementary DNA, map cDNA to reference genome</p>
<div class="columns">
<div class="column" style="width:50%;">
<p><img data-src="Journal.pcbi.1004393.g002.png" alt="&quot;RNA-seq data generation. A typical RNA-seq experimental workflow involves the isolation of RNA from samples of interest, generation of sequencing libraries, use of a high-throughput sequencer to produce hundreds of millions of short paired-end reads, alignment of reads against a reference genome or transcriptome, and downstream analysis for expression estimation, differential expression, transcript isoform discovery, and other applications. Refer to S1 Table, S3 Table, and S7 Table for more details on the concepts depicted in this figure.&quot; Image taken from Figure 2 in Griffith, Malachi; Walker, Jason R.; Spies, Nicholas C.; Ainscough, Benjamin J.; Griffith, Obi L. (2015). &quot;Informatics for RNA Sequencing: A Web Resource for Analysis on the Cloud&quot;. PLOS Computational Biology. 11 (8): e1004393. doi:10.1371/journal.pcbi.1004393. ISSN 1553-7358." width="1100"></p>
<p><span class="citation" data-cites="griffith2015informatics">(<a href="#/thank-you" role="doc-biblioref" onclick="">Griffith et al. 2015</a>)</span></p>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="https://github.com/hbctraining/DGE_workshop_salmon_online/blob/master/lessons/02_DGE_count_normalization.md"><img data-src="normalization_methods_depth.png"></a></p>
</figure>
</div>
<p><span class="citation" data-cites="meeta_mistry_2021_4783481">(<a href="#/thank-you" role="doc-biblioref" onclick="">Mistry et al. 2021</a>)</span></p>
</div>
</div>
</section>
<section id="what-does-the-data-look-like" class="slide level2">
<h2>What does the data look like?</h2>
<p><strong>Raw read sequencing data (large, fastq files)</strong> <span class="math inline">\(\to\)</span> <strong>Count data (gene ID x sample, matrix in .csv)</strong></p>
<p>From a downstream statistical analysis standpoint:</p>
<ul>
<li><p>Level of measurement is the gene (or transcript) or “feature”</p></li>
<li><p>Measurement/outcome is “gene count”: how many reads aligned to that gene’s RNA sequence after alignment to a reference genome? <span class="math inline">\(\to\)</span> count matrix</p></li>
<li><p>Need a statistical model for <strong>count data</strong> that models the variability appropriately</p></li>
</ul>

<img data-src="countdata.png" class="r-stretch quarto-figure-center"><p class="caption"><a href="https://hbctraining.github.io/DGE_workshop_salmon_online/lessons/01c_RNAseq_count_distribution.html">hbctraining</a> <span class="citation" data-cites="meeta_mistry_2021_4783481">Mistry et al. (<a href="#/thank-you" role="doc-biblioref" onclick="">2021</a>)</span></p></section>
<section id="simplest-study-two-groups" class="slide level2">
<h2>Simplest Study: Two groups</h2>
<p>Suppose we are studying two treatment groups (treatment vs.&nbsp;control), and we want to know which genes are deferentially expressed between these two groups.</p>
<p><strong>Two groups</strong>: Two experimental groups, multiple <em>biological</em> replicates within each group. The two groups contain different samples (i.e., not paired, not the same samples over time).</p>
<p><strong>Differentially expressed genes (DEGs)</strong>: A gene is “differentially expressed” between the treatment vs.&nbsp;control if there is a difference observed in read counts or expression levels.</p>
<p><strong>Question of interest:</strong> How many samples (biological replicates) in each group to we need to “detect a difference”?</p>
<ul>
<li>… with some level of confidence (power), restricting false positives (type I error, FDR)?</li>
</ul>
</section>
<section id="simplest-study-two-groups-1" class="slide level2">
<h2>Simplest Study: Two groups</h2>
<p><strong>Question of interest:</strong> How many samples to we need to “detect a difference”?</p>
<p><strong>(Some) Information we need:</strong></p>
<ul>
<li><p>Definition of “how different” (i.e.&nbsp;effect size, fold change)</p></li>
<li><p>Desired level of power and type I error/p-value expectations (“significance level”)</p></li>
<li><p>Information about sequencing depth, average read count</p></li>
<li><p>Gene expression info: expected variability/dispersion of gene expression levels in each group, number of genes we expect to measure/observe</p></li>
<li><p>Statistical model and test, method for multiple comparison adjustment</p></li>
<li><p>“How much signal?” % genes differentially expressed above some fold change</p></li>
</ul>
</section>
<section id="reminder-statistical-power-concepts" class="slide level2 center">
<h2>Reminder: Statistical Power Concepts</h2>
</section>
<section id="sample-size" class="slide level2">
<h2>Sample Size</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>More biological replicates leads to:</p>
<ul>
<li><p>better estimates of variation (gene, biological, sample-to-sample)</p></li>
<li><p>identify outliers or possible sources of technical variation (batch effects)</p></li>
<li><p>improve precision of estimates</p></li>
<li><p>observe low abundance genes</p></li>
</ul>
</div><div class="column" style="width:50%;">
<p><img data-src="techrep.jpg"></p>
</div>
</div>
<p><span class="citation" data-cites="klaus2015statistical">Klaus (<a href="#/thank-you" role="doc-biblioref" onclick="">2015</a>)</span> and <span class="citation" data-cites="meeta_mistry_2021_4783481">Mistry et al. (<a href="#/thank-you" role="doc-biblioref" onclick="">2021</a>)</span></p>
</section>
<section id="components-of-sample-size" class="slide level2">
<h2>Components of Sample Size</h2>
<p>In general, need to know 3 of the 4 to determine the 4th:</p>
<table>
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th>Do We Know?</th>
<th>Measure</th>
<th>Definition</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>??</td>
<td>Effect Size</td>
<td>Magnitude of difference or fold change</td>
</tr>
<tr class="even">
<td>??</td>
<td>Sample Size</td>
<td>N total, n per group</td>
</tr>
<tr class="odd">
<td>0.05, 0.01</td>
<td>Type I Error / Significance level</td>
<td><span class="math inline">\(\alpha\)</span> = probability of rejecting null hypothesis when it is true <em>need to adjust for multiple comparisons!</em></td>
</tr>
<tr class="even">
<td>0.9, 0.8</td>
<td>Power</td>
<td>1 - <span class="math inline">\(\beta\)</span> = 1 - Type II error = probability of rejecting null hypothesis when it is false <em>need to consider multiple comparisons!</em></td>
</tr>
</tbody>
</table>
</section>
<section id="effect-size-mean-expression-ratio-fc" class="slide level2">
<h2>Effect size = Mean Expression Ratio (FC)</h2>
<p>In a two group comparison, effect size is usually <strong><em>fold change</em></strong> (or logFC) for an individual gene:</p>
<p><strong>FC<sub>gene&nbsp;X</sub> = (mean expression of gene X in trt) / (mean expression of gene X in control)</strong></p>
<ul>
<li><p>May observe a large range of fold changes, or fold changes may be close to 1 for all genes</p></li>
<li><p>FC measures the average size of the difference between groups, not variability</p></li>
<li><p>Alternative effect size: biological coefficient of variation (BCV)</p>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="https://github.com/hbctraining/DGE_workshop_salmon_online/blob/master/lessons/06_DGE_visualizing_results.md"><img data-src="volcano_plot_2_salmon.png"></a></p>
<p></p><figcaption>Volcano plot (hbctraining)</figcaption><p></p>
</figure>
</div>
<p><span class="citation" data-cites="meeta_mistry_2021_4783481">(<a href="#/thank-you" role="doc-biblioref" onclick="">Mistry et al. 2021</a>)</span></p></li>
</ul>
</section>
<section id="type-i-error-with-multiple-comparisons" class="slide level2">
<h2>Type I error (with multiple comparisons)</h2>
<p>“The worst error” Truth = No difference, Test conclusion = there is a difference</p>
<ul>
<li><p>Typical power calculations are for one test, we could have tens of thousands of tests for each gene</p></li>
<li><p>Multiple comparisons/multiple testing problem: If we use p-value &lt; 0.05, for one test there is a 5% change we have a false positive, if we test 20,000 genes and use p&lt;0.05, we would expect to detect 1000 false positives by chance; if we found 3000 DEGs total, one third are false positives</p></li>
<li><p>Use “False Discovery Rate” = FDR as “adjusted” p-values</p>
<ul>
<li><p>FDR = proportion of expected false positives in our set of DEGs, often controlled at 5% or 10%</p></li>
<li><p>Benjamini-Hochberg method common, alternative Storey/Q-value method, see <span class="citation" data-cites="meeta_mistry_2021_4783481">(<a href="#/thank-you" role="doc-biblioref" onclick="">Mistry et al. 2021</a>)</span> for summary</p></li>
</ul></li>
</ul>
</section>
<section id="power-with-multiple-comparisons" class="slide level2">
<h2>Power (with multiple comparisons)</h2>
<ul>
<li><p>With one test, power = 1 - type II error for a given effect size</p>
<ul>
<li>Power = probability that our test is “significant” when the truth is that there is a difference</li>
</ul></li>
<li><p>With multiple tests/genes, could have many definitions</p>
<ul>
<li><p>Probability we detect at least X% of genes that are truly different (at least some FC)</p></li>
<li><p>Probability that we detect <em>all</em> genes that are truly different (at least some FC)</p></li>
<li><p><strong>Probability that we detect <em>one specific gene</em> that is truly different by some minimum FC</strong></p></li>
</ul></li>
<li><p>Define “significant” as FDR &lt; cutoff</p></li>
</ul>
</section>
<section id="gene-expressionabundance-distributions" class="slide level2">
<h2>Gene Expression/Abundance Distributions</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li><p>Count data = discrete data (0, 1, 2…)</p></li>
<li><p>Need <strong>probability distribution</strong> to match the type of data, such as: Poisson distribution, Negative Binomial distribution, Poisson-gamma mixture distribution</p></li>
<li><p>Poisson model for count data assumes mean = variance, but RNA-seq has other <strong>sources of variation</strong> than the counting process <span class="math inline">\(\to\)</span> NB with dispersion parameter, other models</p></li>
<li><p>Lower count genes can be harder to detect/measure/test</p></li>
<li><p>Sequencing depth (total number of reads, i.e.&nbsp;5-200 million) influences <strong>average gene count</strong></p></li>
</ul>
</div><div class="column" style="width:50%;">
<p><img data-src="deseq_nb.png"></p>
<p><span class="citation" data-cites="meeta_mistry_2021_4783481">(<a href="#/thank-you" role="doc-biblioref" onclick="">Mistry et al. 2021</a>)</span></p>
</div>
</div>
</section>
<section id="variability" class="slide level2">
<h2>Variability</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><img data-src="meanvar.png"></p>
</div><div class="column" style="width:50%;">
<p><img data-src="de_variation.png"></p>
</div>
</div>
<p><span class="citation" data-cites="meeta_mistry_2021_4783481">(<a href="#/thank-you" role="doc-biblioref" onclick="">Mistry et al. 2021</a>)</span></p>
</section>
<section id="factors-affecting-gene-expression-power" class="slide level2">
<h2>Factors affecting gene expression + power</h2>
<ul>
<li><p><strong><em>Biological differences and variation</em></strong></p></li>
<li><p>Sequencing depth/coverage</p></li>
<li><p>Gene length</p></li>
<li><p>Total sample RNA output, variation between samples within biological groups</p></li>
<li><p>Variance/dispersion of gene abundance distributions</p>
<ul>
<li>dispersion: parameter that defines how far we expect observed count will deviate from mean value (estimate from a model)</li>
</ul></li>
</ul>
</section>
<section id="sequencing-depth-vs.-n" class="slide level2">
<h2>Sequencing depth vs.&nbsp;N</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li><p>In the past with older NGS technology, limited budget might have restricted sequencing depth (fewer total read counts, more samples per lane) if more biological replicates were desired</p></li>
<li><p>Sequencing depth is now “cheaper” so less of an issue, but still keep in mind abundance levels of genes you want to measure/detect</p></li>
<li><p>Low abundance genes are difficult to test/detect since counting variability may be larger than biological variability/differences, may be filtered out when performing QC/low count filtering</p></li>
<li><p>See discussion in <span class="citation" data-cites="liu2014rna">Liu, Zhou, and White (<a href="#/thank-you" role="doc-biblioref" onclick="">2014</a>)</span> and guidelines in <span class="citation" data-cites="meeta_mistry_2021_4783481">Mistry et al. (<a href="#/thank-you" role="doc-biblioref" onclick="">2021</a>)</span> <a href="https://github.com/hbctraining/DGE_workshop_salmon_online/blob/master/lessons/experimental_planning_considerations.md">Experimental Planning Considerations</a> lesson</p></li>
</ul>
</div><div class="column" style="width:50%;">
<ul>
<li><p>Paired-end reads are standard now with 100bp length</p></li>
<li><p><span class="math inline">\(&gt;\)</span> 30 million reads recommended and standard, <span class="math inline">\(&gt;\)</span> 60m for low abundance or novel isoforms</p></li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="https://github.com/hbctraining/DGE_workshop_salmon_online/blob/master/lessons/01c_RNAseq_count_distribution.md"><img data-src="de_replicates_img2.png"></a></p>
</figure>
</div>
<p>[<span class="citation" data-cites="liu2014rna">Liu, Zhou, and White (<a href="#/thank-you" role="doc-biblioref" onclick="">2014</a>)</span>]</p>
</div>
</div>
</section>
<section id="typical-analysis" class="slide level2">
<h2>Typical analysis</h2>
<ul>
<li><p>Common tools: <code>DESeq2</code>, <code>edgeR</code>, <code>baySeq</code>, and <code>limma</code>/<code>voom</code> packages in R/Bioconductor</p></li>
<li><p>Normalization for gene length, sequencing depth (TMM, RPKM)</p></li>
<li><p><code>DESeq2</code>, <code>edgeR</code>, <code>baySeq</code> use generalized linear models (GLMs) with negative binomial (NB) distribution to fit the data:</p>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="https://github.com/hbctraining/DGE_workshop_salmon_online/blob/master/lessons/05a_hypothesis_testing.md"><img data-src="glm.png"></a></p>
</figure>
</div></li>
<li><p><code>Voom</code> models mean-variance relationships to use normal based models in <code>limma</code> (linear model with Empirical Bayes variance smoothing)</p></li>
<li><p>Hypothesis test options: Wald Test, Likelihood Ratio Test, and similar</p></li>
</ul>
</section>
<section id="power-calculation-softwaretools" class="slide level2">
<h2>Power calculation software/tools</h2>
<ul>
<li><p>See review in <span class="citation" data-cites="jeon2023statistical">Jeon et al. (<a href="#/thank-you" role="doc-biblioref" onclick="">2023</a>)</span> and comparisons in <span class="citation" data-cites="poplawski2018feasibility">Poplawski and Binder (<a href="#/thank-you" role="doc-biblioref" onclick="">2018</a>)</span></p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/RNASeqPower.html">RNASeqPower</a> Bioconductor/R package</p></li>
<li><p><a href="https://www.bioconductor.org/packages/release/bioc/html/RnaSeqSampleSize.html">RnaSeqSampleSize</a> Bioconductor/R package and <a href="https://cqs-vumc.shinyapps.io/rnaseqsamplesizeweb/">web/Shiny app</a></p></li>
<li><p><a href="https://cran.r-project.org/web/packages/ssizeRNA/index.html">ssizeRNA</a> R package (based on voom/limma)</p></li>
<li><p><a href="https://bioconductor.org/packages/devel/bioc/vignettes/PROPER/inst/doc/PROPER.pdf">PROPER</a> simulation based tool</p></li>
</ul>
<p>Based on above reviews, ssizeRNA and PROPER were most recommended</p>
</section>
<section id="example-r-code-ssizerna" class="slide level2">
<h2>Example R code, ssizeRNA</h2>
<p>Adapted from <code>ssizeRNA</code> <a href="https://cran.r-project.org/web/packages/ssizeRNA/vignettes/ssizeRNA.pdf">vignette</a> and Suppl Material S1 <span class="citation" data-cites="jeon2023statistical">Jeon et al. (<a href="#/thank-you" role="doc-biblioref" onclick="">2023</a>)</span></p>
<p>Arguments for <code>ssize_single()</code> function, assuming all genes have the same characteristics:</p>
<ul>
<li>Total number of genes: <code>G = 10000</code></li>
<li>Proportion of non-DE “null” genes: <code>pi0 = 0.8</code></li>
<li>FDR level to control: <code>fdr = 0.05</code></li>
<li>Desired average power to achieve: <code>power = 0.8</code></li>
<li>Average read count for each gene in control group: <code>mu = 10</code></li>
<li>Dispersion parameter for each gene: <code>disp = 0.1</code></li>
<li>Fold change for each gene: <code>fc = 2</code></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="do">## Install ssizeRNA package if needed</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="co"># install.packages("ssizeRNA")</span></span>
<span id="cb1-3"><a href="#cb1-3"></a></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="do">## Load package</span></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(ssizeRNA)</span>
<span id="cb1-6"><a href="#cb1-6"></a></span>
<span id="cb1-7"><a href="#cb1-7"></a>size_out <span class="ot">&lt;-</span> <span class="fu">ssizeRNA_single</span>(<span class="at">nGenes =</span> <span class="dv">10000</span>, </span>
<span id="cb1-8"><a href="#cb1-8"></a>                      <span class="at">pi0 =</span> <span class="fl">0.8</span>,</span>
<span id="cb1-9"><a href="#cb1-9"></a>                      <span class="at">fdr =</span> <span class="fl">0.05</span>,</span>
<span id="cb1-10"><a href="#cb1-10"></a>                      <span class="at">power =</span> <span class="fl">0.8</span>,</span>
<span id="cb1-11"><a href="#cb1-11"></a>                      <span class="at">mu =</span> <span class="dv">10</span>,</span>
<span id="cb1-12"><a href="#cb1-12"></a>                      <span class="at">disp =</span> <span class="fl">0.1</span>,</span>
<span id="cb1-13"><a href="#cb1-13"></a>                      <span class="at">fc =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="co"># Output variables:</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="co"># ssize: sample sizes (for each treatment) at which desired power is first reached. size$ssize</span></span>
<span id="cb2-3"><a href="#cb2-3"></a>size_out<span class="sc">$</span>ssize</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     pi0 ssize     power
[1,] 0.8    14 0.8393099</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># power: power calculations with corresponding sample sizes.</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>size_out<span class="sc">$</span>power</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       n       0.8
 [1,]  3 0.0000000
 [2,]  4 0.0000000
 [3,]  5 0.0061645
 [4,]  6 0.0984413
 [5,]  7 0.2381532
 [6,]  8 0.3756802
 [7,]  9 0.4959341
 [8,] 10 0.5962760
 [9,] 11 0.6781612
[10,] 12 0.7442056
[11,] 13 0.7971129
[12,] 14 0.8393099
[13,] 15 0.8728672
[14,] 16 0.8995036
[15,] 17 0.9206182
[16,] 18 0.9373283
[17,] 19 0.9505492
[18,] 20 0.9609931
[19,] 21 0.9692400
[20,] 22 0.9757500
[21,] 23 0.9808852
[22,] 24 0.9849333
[23,] 25 0.9881243
[24,] 26 0.9906379
[25,] 27 0.9926184
[26,] 28 0.9941786
[27,] 29 0.9954073
[28,] 30 0.9963752
[29,] 31 0.9971380
[30,] 32 0.9977389
[31,] 33 0.9982126
[32,] 34 0.9985861
[33,] 35 0.9988806</code></pre>
</div>
</div>
<img data-src="2023_11_pss_rnaseq_files/figure-revealjs/unnamed-chunk-1-1.png" width="960" class="r-stretch"></section>
<section id="typical-inputs" class="slide level2">
<h2>Typical inputs</h2>
<ul>
<li>Dispersion often depends on animal model, might be 0.1 for mouse data, 0.2-0.5 for human data</li>
<li>Read count depends on sequencing depth and distribution of abundance, if using average can be low, typically 5-30, but in reality will vary widely by gene</li>
<li>Proportion of non-DEG, varies greatly based on experiment/biological difference expectations</li>
<li>Fold change, the magic number :) 1.5, 2, 2.5…</li>
<li>Power 0.9 and FDR 0.05, more “relaxed” adjust power 0.8-0.85 and/or FDR 0.1</li>
</ul>
</section>
<section id="interactive-web-app" class="slide level2">
<h2>Interactive web app</h2>
<ul>
<li><a href="https://www.bioconductor.org/packages/release/bioc/html/RnaSeqSampleSize.html">RnaSeqSampleSize</a> Bioconductor/R package and <a href="https://cqs-vumc.shinyapps.io/rnaseqsamplesizeweb/">web/Shiny app</a></li>
</ul>
<div class="columns">
<div class="column" style="width:50%;">
<p><img data-src="webapp1.png"></p>
</div><div class="column" style="width:50%;">
<p><img data-src="webapp2.png"></p>
</div>
</div>
</section>
<section id="pilot-data-or-public-data" class="slide level2">
<h2>Pilot data or public data</h2>
<ul>
<li>Ideally, pilot data specific to the experiment would be used to determine expected variability/dispersion and effect sizes</li>
<li><code>ssizeRNA</code> gives examples on using existing data to estimate <code>mu</code> and <code>disp</code> across all the genes, then use <code>ssizeRNA_vary()</code> function, and see below from Suppl Material S1 <span class="citation" data-cites="jeon2023statistical">Jeon et al. (<a href="#/thank-you" role="doc-biblioref" onclick="">2023</a>)</span> for R code</li>
<li><code>RnaSeqSampleSize</code> has TCGA data examples, see <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/RnaSeqSampleSize/inst/doc/RnaSeqSampleSize.pdf">vignette</a>, and also <a href="https://cqs-vumc.shinyapps.io/rnaseqsamplesizeweb/">web app</a></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="do">## Install other packages if needed.</span></span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="co"># install.packages("BiocManager")</span></span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="co"># BiocManager::install("edgeR") # cf. https://bioconductor.org/packages/release/bioc/html/edge R.html</span></span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="co"># BiocManager::install("Biobase") # cf. https://bioconductor.org/packages/release/bioc/html/Biob ase.html</span></span>
<span id="cb6-5"><a href="#cb6-5"></a></span>
<span id="cb6-6"><a href="#cb6-6"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb6-7"><a href="#cb6-7"></a><span class="fu">library</span>(Biobase)</span>
<span id="cb6-8"><a href="#cb6-8"></a><span class="fu">library</span>(ssizeRNA)</span>
<span id="cb6-9"><a href="#cb6-9"></a></span>
<span id="cb6-10"><a href="#cb6-10"></a><span class="do">## Example data saved in ssizeRNA package:</span></span>
<span id="cb6-11"><a href="#cb6-11"></a><span class="do">## Step 2-1. load hammer dataset (Hammer, P. et al., 2010)</span></span>
<span id="cb6-12"><a href="#cb6-12"></a><span class="do">## two group rat data</span></span>
<span id="cb6-13"><a href="#cb6-13"></a><span class="fu">data</span>(hammer.eset)</span>
<span id="cb6-14"><a href="#cb6-14"></a>counts <span class="ot">&lt;-</span> <span class="fu">exprs</span>(hammer.eset)[,<span class="fu">phenoData</span>(hammer.eset)<span class="sc">$</span>Time<span class="sc">==</span><span class="st">"2 weeks"</span>]</span>
<span id="cb6-15"><a href="#cb6-15"></a>counts <span class="ot">&lt;-</span> counts[<span class="fu">rowSums</span>(counts) <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb6-16"><a href="#cb6-16"></a><span class="fu">dim</span>(counts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 18463     4</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>trt <span class="ot">&lt;-</span> hammer.eset<span class="sc">$</span>protocol[<span class="fu">which</span>(hammer.eset<span class="sc">$</span>Time<span class="sc">==</span><span class="st">"2 weeks"</span>)]</span>
<span id="cb8-2"><a href="#cb8-2"></a></span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="do">## After generating count data with column names of control and treatment, you may estimate the parameters of mu and disp using the following script.</span></span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="do">## mu: average read count in the control group</span></span>
<span id="cb8-5"><a href="#cb8-5"></a><span class="do">## The following apply function averages the count values for each gene in the control group.</span></span>
<span id="cb8-6"><a href="#cb8-6"></a>mu <span class="ot">&lt;-</span> <span class="fu">apply</span>(counts[, trt <span class="sc">==</span> <span class="st">"control"</span>], <span class="dv">1</span>, mean)</span>
<span id="cb8-7"><a href="#cb8-7"></a><span class="fu">summary</span>(mu)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
     0.0      9.5    146.5    952.0    662.5 386345.0 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="do">## disp: dispersion parameters estimates using the edgeR package with count data.</span></span>
<span id="cb10-2"><a href="#cb10-2"></a>d <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(counts)</span>
<span id="cb10-3"><a href="#cb10-3"></a>d <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(d)</span>
<span id="cb10-4"><a href="#cb10-4"></a>d <span class="ot">&lt;-</span> <span class="fu">estimateCommonDisp</span>(d) </span>
<span id="cb10-5"><a href="#cb10-5"></a>d <span class="ot">&lt;-</span> <span class="fu">estimateTagwiseDisp</span>(d) </span>
<span id="cb10-6"><a href="#cb10-6"></a>disp <span class="ot">&lt;-</span> d<span class="sc">$</span>tagwise.dispersion</span>
<span id="cb10-7"><a href="#cb10-7"></a><span class="fu">summary</span>(disp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.05566 0.06991 0.11093 0.13949 0.18397 1.01143 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>size_out <span class="ot">&lt;-</span> <span class="fu">ssizeRNA_vary</span>(<span class="at">nGenes =</span> <span class="dv">10000</span>, </span>
<span id="cb12-2"><a href="#cb12-2"></a>                         <span class="at">mu =</span> mu, <span class="co"># estimated above</span></span>
<span id="cb12-3"><a href="#cb12-3"></a>                         <span class="at">disp =</span> disp, <span class="co"># estimated above</span></span>
<span id="cb12-4"><a href="#cb12-4"></a>                         <span class="at">fc =</span> <span class="dv">2</span>,</span>
<span id="cb12-5"><a href="#cb12-5"></a>                         <span class="at">up =</span> <span class="fl">0.5</span>,</span>
<span id="cb12-6"><a href="#cb12-6"></a>                         <span class="at">fdr =</span> <span class="fl">0.05</span>,</span>
<span id="cb12-7"><a href="#cb12-7"></a>                         <span class="at">power =</span> <span class="fl">0.8</span>,</span>
<span id="cb12-8"><a href="#cb12-8"></a>                         <span class="at">maxN =</span> <span class="dv">35</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co"># Output variables:</span></span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="co"># ssize: sample sizes (for each treatment) at which desired power is first reached. size$ssize</span></span>
<span id="cb13-3"><a href="#cb13-3"></a>size_out<span class="sc">$</span>ssize</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     pi0 ssize     power
[1,] 0.8     8 0.8068609</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="co"># power: power calculations with corresponding sample sizes.</span></span>
<span id="cb15-2"><a href="#cb15-2"></a>size_out<span class="sc">$</span>power</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       n       0.8
 [1,]  3 0.0000000
 [2,]  4 0.1584711
 [3,]  5 0.4421728
 [4,]  6 0.6214126
 [5,]  7 0.7337198
 [6,]  8 0.8068609
 [7,]  9 0.8562598
 [8,] 10 0.8906682
 [9,] 11 0.9152706
[10,] 12 0.9332618
[11,] 13 0.9466712
[12,] 14 0.9568361
[13,] 15 0.9646599
[14,] 16 0.9707612
[15,] 17 0.9755790
[16,] 18 0.9794242
[17,] 19 0.9825252
[18,] 20 0.9850497
[19,] 21 0.9871208
[20,] 22 0.9888359
[21,] 23 0.9902645
[22,] 24 0.9914644
[23,] 25 0.9924768
[24,] 26 0.9933376
[25,] 27 0.9940732
[26,] 28 0.9947049
[27,] 29 0.9952501
[28,] 30 0.9957233
[29,] 31 0.9961354
[30,] 32 0.9964963
[31,] 33 0.9968129
[32,] 34 0.9970923
[33,] 35 0.9973394</code></pre>
</div>
</div>
<img data-src="2023_11_pss_rnaseq_files/figure-revealjs/unnamed-chunk-2-1.png" width="960" class="r-stretch"></section>
<section id="more-complex-studies" class="slide level2">
<h2>More complex studies</h2>
<p>What are the additional experimental factors?</p>
<ul>
<li><p>Multiple groups, multiple treatment combinations</p></li>
<li><p>Interaction effects</p></li>
<li><p>Continuous predictor rather than groups</p></li>
<li><p>Time or repeated measure component</p></li>
<li><p>Technical replicates</p></li>
<li><p>Batch effects or technical variation</p></li>
</ul>
<p><span class="math inline">\(\to\)</span> Likely need to use <strong>simulation based methods</strong> for power/sample size</p>
<h3 id="pathway-analyses">Pathway analyses?</h3>
<p>Even more complex to calculate power with simulations and many assumptions</p>
</section>
<section id="single-cell-rna-seq-or-spatial-transcriptomics" class="slide level2">
<h2>Single-cell RNA-seq or spatial transcriptomics</h2>

<img data-src="scrnacomparison.png" class="r-stretch"><p><span class="citation" data-cites="jeon2023statistical">(<a href="#/thank-you" role="doc-biblioref" onclick="">Jeon et al. 2023</a>)</span></p>
</section>
<section id="scrna-seq" class="slide level2">
<h2>scRNA-seq</h2>
<ul>
<li><p>scRNA-seq have more 0 counts than bulk which affects dispersions/variation, require “zero-inflated” model</p></li>
<li><p>Power analysis for DEG (1) between biological groups/experimental conditions for a specific cell type <em>or</em> (2) between cell types or sub-populations within a sample</p></li>
<li><p>Power analyses for identifying cell sub-populations, either within a single tissue, or proportional differences across experimental conditions</p>
<p><img data-src="scrna_powertools.png"></p>
<p><span class="citation" data-cites="jeon2023statistical">(<a href="#/thank-you" role="doc-biblioref" onclick="">Jeon et al. 2023</a>)</span></p></li>
</ul>
</section>
<section id="conclusions" class="slide level2">
<h2>Conclusions</h2>
<h3 id="important-factors-bulk-rna-seq-deg">Important Factors (bulk RNA-seq DEG)</h3>
<ul>
<li><p>Distributions: average (or distribution of) read counts, as well as average (or distribution of, or maximum) dispersion</p></li>
<li><p>Proportion of differentially expressed genes</p></li>
<li><p>Fold change effect size</p></li>
<li><p>False discovery rate and power</p></li>
<li><p>Experimental design complexity (beyond two groups, consult statistician)</p></li>
</ul>
<h3 id="primary-resources">Primary resources</h3>
<ul>
<li><span class="citation" data-cites="jeon2023statistical">Jeon et al. (<a href="#/thank-you" role="doc-biblioref" onclick="">2023</a>)</span> and <span class="citation" data-cites="meeta_mistry_2021_4783481">Mistry et al. (<a href="#/thank-you" role="doc-biblioref" onclick="">2021</a>)</span> <a href="https://github.com/hbctraining/DGE_workshop_salmon_online/tree/master/lessons">tutorial</a></li>
</ul>
</section>
<section id="thank-you" class="slide level2 smaller scrollable">
<h2>Thank you</h2>
<ul>
<li><p>Recording will be available at <a href="https://www.ohsu.edu/octri/octri-research-forum-your-monthly-clinical-and-translational-research-event">OCTRI BERD Research Forum Website</a></p></li>
<li><p>email: minnier [at] ohsu.edu</p></li>
<li><p>Thanks for consultation/tips from ONPRC BBC Director Suzi Fei and all I’ve learned from IGL/MPSSR Directors Bob Searles and Chris Harrington</p></li>
<li><p>Please fill out the survey that Amy Laird will send you!</p></li>
<li><p>More “omics” PSS series to come</p></li>
<li><p>Need stats help? Contact <a href="https://www.ohsu.edu/research-cores/biostatistics-design-program">BDP</a> or <a href="https://www.ohsu.edu/knight-cancer-institute/biostatistics-shared-resource">Knight BSR</a></p></li>
</ul>
<h3 id="references">References</h3>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-conesa2016survey" class="csl-entry" role="doc-biblioentry">
Conesa, Ana, Pedro Madrigal, Sonia Tarazona, David Gomez-Cabrero, Alejandra Cervera, Andrew McPherson, Michał Wojciech Szcześniak, et al. 2016. <span>“A Survey of Best Practices for RNA-Seq Data Analysis.”</span> <em>Genome Biology</em> 17 (1): 1–19.
</div>
<div id="ref-griffith2015informatics" class="csl-entry" role="doc-biblioentry">
Griffith, Malachi, Jason R Walker, Nicholas C Spies, Benjamin J Ainscough, and Obi L Griffith. 2015. <span>“Informatics for RNA Sequencing: A Web Resource for Analysis on the Cloud.”</span> <em>PLoS Computational Biology</em> 11 (8): e1004393.
</div>
<div id="ref-jeon2023statistical" class="csl-entry" role="doc-biblioentry">
Jeon, Hyeongseon, Juan Xie, Yeseul Jeon, Kyeong Joo Jung, Arkobrato Gupta, Won Chang, and Dongjun Chung. 2023. <span>“Statistical Power Analysis for Designing Bulk, Single-Cell, and Spatial Transcriptomics Experiments: Review, Tutorial, and Perspectives.”</span> <em>Biomolecules</em> 13 (2): 221.
</div>
<div id="ref-klaus2015statistical" class="csl-entry" role="doc-biblioentry">
Klaus, Bernd. 2015. <span>“Statistical Relevance—Relevant Statistics, Part i.”</span> <em>The EMBO Journal</em> 34 (22): 2727–30.
</div>
<div id="ref-lee2002power" class="csl-entry" role="doc-biblioentry">
Lee, Mei-Ling Ting, and George Alex Whitmore. 2002. <span>“Power and Sample Size for DNA Microarray Studies.”</span> <em>Statistics in Medicine</em> 21 (23): 3543–70.
</div>
<div id="ref-liu2014rna" class="csl-entry" role="doc-biblioentry">
Liu, Yuwen, Jie Zhou, and Kevin P White. 2014. <span>“RNA-Seq Differential Expression Studies: More Sequence or More Replication?”</span> <em>Bioinformatics</em> 30 (3): 301–4.
</div>
<div id="ref-meeta_mistry_2021_4783481" class="csl-entry" role="doc-biblioentry">
Mistry, Meeta, Mary Piper, Jihe Liu, and Radhika Khetani. 2021. <span>“<span class="nocase">hbctraining/DGE_workshop_salmon_online: Differential Gene Expression Workshop Lessons from HCBC (first release)</span>.”</span> Zenodo.
</div>
<div id="ref-poplawski2018feasibility" class="csl-entry" role="doc-biblioentry">
Poplawski, Alicia, and Harald Binder. 2018. <span>“Feasibility of Sample Size Calculation for RNA-Seq Studies.”</span> <em>Briefings in Bioinformatics</em> 19 (4): 713–20.
</div>
</div>
<div class="footer footer-default">
<p><a href="https://bit.ly/berd_pss_rnaseq" class="uri">https://bit.ly/berd_pss_rnaseq</a></p>
</div>
</section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="2023_11_pss_rnaseq_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="2023_11_pss_rnaseq_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="2023_11_pss_rnaseq_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="2023_11_pss_rnaseq_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="2023_11_pss_rnaseq_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="2023_11_pss_rnaseq_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="2023_11_pss_rnaseq_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="2023_11_pss_rnaseq_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="2023_11_pss_rnaseq_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="2023_11_pss_rnaseq_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': true,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': true,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: true,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1280,

        height: 720,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        target: function(trigger) {
          return trigger.previousElementSibling;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'quarto-reveal',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>